cvat-ui/src/components/login-page/login-page.tsx

```diff
@@ -22,16 +24,58 @@ interface LoginPageComponentProps {
     socialAuthMethods: SocialAuthMethods;
     onLogin: (credential: string, password: string) => void;
     loadSocialAuthenticationMethods: () => void;
+    loadSSOConfiguration: () => void;
 }
 
-const renderSocialAuthMethods = (methods: SocialAuthMethods): JSX.Element[] => {
+const renderSocialAuthMethods = (methods: SocialAuthMethods): JSX.Element | JSX.Element[] => {
     const { backendAPI } = cvat.config;
+    const activeMethods = methods.filter((item: SocialAuthMethod) => item.isEnabled);
+
+    if (activeMethods.length > 2) {
+        const numberPerRow = 4;
+        const map: SocialAuthMethod[][] = [];
+
+        activeMethods.forEach((el) => {
+            if (!map.length || map[map.length - 1].length === numberPerRow) {
+                map.push([el]);
+            } else {
+                map[map.length - 1].push(el);
+            }
+        });
+
+        return (
+            <>
+                {map.map((item: SocialAuthMethods, idx): JSX.Element => (
+                    <Row
+                        gutter={[4, 4]}
+                        align='middle'
+                        justify='space-between'
+                        key={idx}
+                        className='cvat-social-authentication-row-with-icons'
+                    >
+                        {item.map((method: SocialAuthMethod) => (
+                            <Col span={6} key={method.provider} style={{ display: 'flex' }}>
+                                <SocialAccountCard
+                                    key={method.provider}
+                                    icon={method.icon || method.publicName}
+                                    href={(method.provider !== 'sso') ? `${backendAPI}/auth/${method.provider}/login/` : '/auth/sso/select-identity-provider/'}
```

Comment:
Lets create enum for method.provider values

Reply:
We cannot use the enum here because the server can be configured with different providers. I've added SSO_PROVIDER_KEY